### PR TITLE
Add BaseMap Gallery To Sidebar

### PR DESCRIPTION
**Pull request by pinnockf**
Screenshot 2024-09-13 at 3 37 44 PM

### CODE REVIEW

File: jtflask/static/assets/css/sidebar.css [hunk 1]
```diff
@@ -97,7 +97,9 @@
 }
 
 .sidebar-main {
-
+    display: flex;
+    flex-direction: column;
+    gap: 20px;
```
Comment (pinnockf) [inline jtflask/static/assets/css/sidebar.css:102]: fixed the styling on the sidebar

File: jtflask/jtdash/sidebar.py [hunk 2]
```diff
@@ -146,14 +125,17 @@ def sidebar(
...
         [
-            html.Div(
-                [sidebar_brand, sidebar_main, collapse_button_container],
-                id="sidebar-col1",
-            ),
-            scrollable_div_drawer,
+            sidebar_left_column,
+            sidebar_drawer,
             dcc.Store(
                 id="drawer-content-store",
```
Comment (pinnockf) [inline jtflask/jtdash/sidebar.py:140]: reduced complexity in sidebar.py

File: jtflask/jtdash/sidebar.py [hunk 3]
```diff
@@ -33,56 +24,53 @@ def get_sidebar_components():
...
+            dmc.Button(
+                id="charts-toggle-button",
+                leftSection=get_icon("fluent:data-pie-20-regular"),
                 className="sidebar-icon",
                 **{"data-position": "center"},
             ),
-            dmc.NavLink(
-                id="open-arcgis-drawer-link",
+            dmc.Button(
+                id="arcgis-tools-toggle-button",
```
Comment (pinnockf) [inline jtflask/jtdash/sidebar.py:38]: Changed from NavLink to Button since we are not navigating

File: jtflask/jtdash/layout.py [hunk 4]
```diff
@@ -2,17 +2,9 @@
...
 from .statshovercards import stats_hover_card
 
-# Initialize sidebar components and store as globals
-(
-    global_sidebar_brand,
-    global_sidebar_main_container,
-    global_collapse_button_container,
-    global_scrollable_div_charts,
-    global_scrollable_div_tools,
-) = get_sidebar_components()
```
Comment (pinnockf) [inline jtflask/jtdash/layout.py:15]: removed global and reduced complexity in preparation for using Appshell

File: jtflask/jtdash/callbacks.py [hunk 5]
```diff
@@ -89,3 +32,48 @@ def select_value(value):
...
+                    'basemaps-gallery-toggle-button'])
+        outputs = list(outputs)
+
+        charts_class_name = 'charts-toggle-button slide-in' if all(n_clicks == None for n_clicks in inputs) else outputs[0]
+
+        return {
+            'charts_class_name': charts_class_name,
+            'tools_className': outputs[1],
+            'basemap_gallery_className': outputs[6],
+        }
```
Comment (pinnockf) [inline jtflask/jtdash/callbacks.py:79]: Simplified callback for sidebar button clicks

File: jtflask/jtdash/basemap_gallery.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        ],
+    )
+
+    basemap_gallery = html.Div(
+        id="drawer-basemap-gallery",
+        className="scrollable-div hidden",
+        children=[basemap_gallery_card],
+    )
+    
+    return basemap_gallery
```
Comment (pinnockf) [inline jtflask/jtdash/basemap_gallery.py:23]: moved basemap gallery to its own file

File: jtflask/jtdash/__init__.py [hunk 7]
```diff
@@ -4,7 +4,6 @@
 
 from .layout import layout, html_layout
 from .callbacks import register_callbacks
-from .sidebar import sidebar, get_sidebar_components
```
Comment (pinnockf) [inline jtflask/jtdash/__init__.py:7]: unused import

File: jtflask/jtdash/charts.py [hunk 8]
```diff
@@ -185,15 +181,18 @@ def create_charts():
...
-            className="cardChart"
-        )
+            className="cardChart",
+        ),
     ]
-    return charts
+
+    return html.Div(
+        children=charts, id="chart_scrollable_div", className="scrollable-div"
+    )
```
Comment (pinnockf) [inline jtflask/jtdash/charts.py:198]: returned that actual div for the charts

**Comment by chjch** [comment]
review completed on all changes and changes were made to the main branch. Close.

**Pull request closed without merging**